src/wasm.h

```diff
@@ -1539,86 +657,30 @@ class Module {
 public:
   Module() {};
 
-  FunctionType* getFunctionType(Name name) { assert(functionTypesMap.count(name)); return functionTypesMap[name]; }
-  Import* getImport(Name name) { assert(importsMap.count(name)); return importsMap[name]; }
-  Export* getExport(Name name) { assert(exportsMap.count(name)); return exportsMap[name]; }
-  Function* getFunction(Name name) { assert(functionsMap.count(name)); return functionsMap[name]; }
-  Global* getGlobal(Name name) { assert(globalsMap.count(name)); return globalsMap[name]; }
-
-  FunctionType* checkFunctionType(Name name) { if (!functionTypesMap.count(name)) return nullptr; return functionTypesMap[name]; }
-  Import* checkImport(Name name) { if (!importsMap.count(name)) return nullptr; return importsMap[name]; }
-  Export* checkExport(Name name) { if (!exportsMap.count(name)) return nullptr; return exportsMap[name]; }
-  Function* checkFunction(Name name) { if (!functionsMap.count(name)) return nullptr; return functionsMap[name]; }
-  Global* checkGlobal(Name name) { if (!globalsMap.count(name)) return nullptr; return globalsMap[name]; }
-
-  void addFunctionType(FunctionType* curr) {
-    assert(curr->name.is());
-    functionTypes.push_back(std::unique_ptr<FunctionType>(curr));
-    assert(functionTypesMap.find(curr->name) == functionTypesMap.end());
-    functionTypesMap[curr->name] = curr;
-  }
-  void addImport(Import* curr) {
-    assert(curr->name.is());
-    imports.push_back(std::unique_ptr<Import>(curr));
-    assert(importsMap.find(curr->name) == importsMap.end());
-    importsMap[curr->name] = curr;
-  }
-  void addExport(Export* curr) {
-    assert(curr->name.is());
-    exports.push_back(std::unique_ptr<Export>(curr));
-    assert(exportsMap.find(curr->name) == exportsMap.end());
-    exportsMap[curr->name] = curr;
-  }
-  void addFunction(Function* curr) {
-    assert(curr->name.is());
-    functions.push_back(std::unique_ptr<Function>(curr));
-    assert(functionsMap.find(curr->name) == functionsMap.end());
-    functionsMap[curr->name] = curr;
-  }
-  void addGlobal(Global* curr) {
-    assert(curr->name.is());
-    globals.push_back(std::unique_ptr<Global>(curr));
-    assert(globalsMap.find(curr->name) == globalsMap.end());
-    globalsMap[curr->name] = curr;
-  }
+  FunctionType* getFunctionType(Name name);
+  Import* getImport(Name name);
+  Export* getExport(Name name);
+  Function* getFunction(Name name);
+  Global* getGlobal(Name name);
 
-  void addStart(const Name& s) {
-    start = s;
-  }
+  FunctionType* checkFunctionType(Name name);
```

Comment:
same here, feels like more comments are needed when the meaning isn't right there